src/ui-schema.ts

```diff
@@ -152,6 +152,15 @@ export type OneOfControlOptions = {
 
 export type AnyOfControlOptions = OneOfControlOptions
 
+export const EnumOptions = ["dropdown", "radio", "segmented"] as const
+
+export type EnumOption = (typeof EnumOptions)[number]
+
+export type EnumControlOptions = {
+  optionType?: EnumOption
+  enumValueToLabelMap?: Record<string, string>
```

Comment:
There's probably a way to inject the enum values here and type map's keys as such, but not worth digging into on this first iteration